### PR TITLE
Reduced order models: AeroDisk and Simplified-ElastoDyn (SED)

### PR DESCRIPTION
**Pull request by andrew-platt**
This PR is *NOT* ready for merging.

**Feature or improvement description**
Two new modules are added for low fidelity modeling of aerodynamics and structure.  These modules are intended for use as very low fidelity turbines in a FAST.Farm simulation in order to speed up the overall simulation.  In such a use case, these might be placed in the front few rows of a FAST.Farm simulation where only the turbines in rows further back are of interest, but the wake effects from the leading turbines are needed.

1. AeroDisk -- This module provides a very simple disk actuator model for the aerodynamics as an alternative to AeroDyn 15 and AeroDyn 14.
2. Simplified-ElastoDyn (SED) -- This module provides a very simple 2 DOF elastic model that only includes yaw and generator speed.  This module is not compatible with SubDyn, HydroDyn, or any mooring codes.

**Impacted areas of the software**

- glue code changes
    - new mesh mappings are introduced to map loads and motions from ADsk and SED to other modules
    - additional logic to accommodate the new modules
    - minor code restructuring as needed
    - checks to ensure SED and ADsk are not used with certain combinations that are not allowed
- FAST.Farm changes
    - Minor updates to FAST_Wrapper to pass correct Cp and Ct values

**Additional supporting information**

**Test results, if applicable**
Three new OpenFAST test cases are added as modifications of the `5MW_Land_DLL_WTurb` case:

- `5MW_Land_DLL_WTurb_SED` -- SED instead of ED
- `5MW_Land_DLL_WTurb_ADsk` -- AeroDisk module instead of AD15
- `5MW_Land_DLL_WTurb_ADsk_SED` -- AeroDisk instead of AD15, and SED instead of ED

New module tests are also included for SED and ADsk.

**To complete**

- [x] Verify correct SED and ADsk behaviour after merging dev
- [x] ~~Phase shift in GenPwr for `5MW_Land_DLL_WTurb_ADsk_SED` case (possibly mesh issue in glue code?)~~
        This was an input file issue
- [x] ~~SED + AD15 can result in tower strike event (possible mesh issue, or maybe setup -- not investigated yet)~~
        This was a setup issue
- [x] Disable blade flexibility in ElastoDyn when ADsk is used with ED
- [x] Add VS build projects for ADsk and SED drivers
- [x] Add a simple FAST.Farm case with a simplified turbine
- [x] add OpenMP directive in AWAE module (high res)
- [ ] Add plots here illustrating comparison to `5MW_Land_DLL_WTurb` case (AD15 + ED with controller)
- [ ] update api_change.rst

### CODE REVIEW

File: modules/simple-elastodyn/src/SED_IO.f90 [hunk 1]
```diff
@@ -0,0 +1,483 @@
+!**********************************************************************************************************************************
+! LICENSING
+! Copyright (C) 2020  National Renewable Energy Laboratory
```
Comment (deslaughter) [inline modules/simple-elastodyn/src/SED_IO.f90:3]:
```suggestion
! Copyright (C) 2024  National Renewable Energy Laboratory
```

File: modules/simple-elastodyn/src/driver/SED_Driver.f90 [hunk 2]
```diff
@@ -0,0 +1,373 @@
+!**********************************************************************************************************************************
+!> ## SED_DriverCode: This code tests the SED module
+!!..................................................................................................................................
+!! LICENSING
+!! Copyright (C) 2012, 2015  National Renewable Energy Laboratory
```
Comment (deslaughter) [inline modules/simple-elastodyn/src/driver/SED_Driver.f90:5]:
```suggestion
!! Copyright (C) 2024  National Renewable Energy Laboratory
```

File: reg_tests/executeAerodiskRegressionCase.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2017 National Renewable Energy Laboratory
```
Comment (deslaughter) [inline reg_tests/executeAerodiskRegressionCase.py:2]:
```suggestion
# Copyright 2024 National Renewable Energy Laboratory
```

File: reg_tests/executeAerodiskRegressionCase.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+#    os.makedirs(testBuildDirectory, exist_ok = True)
+#    for file in glob.glob(os.path.join(inputsDirectory,"*inp")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
+#    for file in glob.glob(os.path.join(inputsDirectory,"*dat")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
+#    for file in glob.glob(os.path.join(inputsDirectory,"*csv")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
```
Comment (deslaughter) [inline reg_tests/executeAerodiskRegressionCase.py:103]:
```suggestion
```

File: reg_tests/executeSimpleElastodynRegressionCase.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2017 National Renewable Energy Laboratory
```
Comment (deslaughter) [inline reg_tests/executeSimpleElastodynRegressionCase.py:2]:
```suggestion
# Copyright 2024 National Renewable Energy Laboratory
```

File: reg_tests/executeSimpleElastodynRegressionCase.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+#    os.makedirs(testBuildDirectory, exist_ok = True)
+#    for file in glob.glob(os.path.join(inputsDirectory,"*inp")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
+#    for file in glob.glob(os.path.join(inputsDirectory,"*dat")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
+#    for file in glob.glob(os.path.join(inputsDirectory,"*csv")):
+#        filename = file.split(os.path.sep)[-1]
+#        shutil.copy(os.path.join(inputsDirectory,filename), os.path.join(testBuildDirectory,filename))
```
Comment (deslaughter) [inline reg_tests/executeSimpleElastodynRegressionCase.py:103]:
```suggestion
```

**Review by deslaughter — APPROVED** [review]

**Comment by andrew-platt** [comment]
TSinflow_126s
Figure 1: XY slice at hub-height with case _TSinflow_ (two 5MW turbines, both with _AeroDyn_ and _ElastoDyn_).

TSinflowADskSED_126s
Figure 2: XY slice at hub-height with case _TSinflowADskSED_ (the leading turbine from the _TSinflow_ case is replaced with the reduced order model using _AeroDisk_ and _SED_).  The results in the wake hitting turbine 2 is very similar resulting in loads on turbine 2 that are very similar to when the leading turbine uses the full _AeroDyn_ + _ElastoDyn_ model.

**Pull request merged**